### PR TITLE
Add Who is online Toast provider and removed notifcations

### PR DESCRIPTION
**Pull request by iszmais**
This is a further improvement to https://github.com/ILIAS-eLearning/ILIAS/pull/5518.
Since WIO Notifications are atm recreated and removed on every page reload they create kind of a mess within the notification Center handling, due to the fact they have a special mode onyl to handle their individual behavior.

Since Those behaviour doesnt match with the common ILIAS Notification and due to the improvement of the new ILIAS Toast feature this PR Proposes to remove the WIO feature from the Notification and make its presentation toast only.

That improves not only the Notification System but also refactores the ancient Fethcer of WIO Notification an put the into the more fitting, non-persisten Toast scope.

### CODE REVIEW

**Comment by iszmais** [comment]
Hello @alex40724 ,

Im not aware of your scedule but i would highly appriate if this could go into the ILIAS 8 release to start the new Version with the improved user view and a clean Database.
Thanks in Advance!

Greetings,
@iszmais

**Comment by alex40724** [comment]
Hi @iszmais ,
thanks for the PR. I have difficulties to understand the full impact. ILIAS 7 was old OSD only, right? With your changes from March 2022 this became notification center + toast? And now it should be toast only, right? I currently checked the ILIAS 8 test installation and neither got a toast or a notification, even if the users newly appeared in the who-is-online list. I think we should get JourFixe confirmation on the expected behaviour here, I set the label.

**Comment by mjansenDatabay** [comment]
Hi @alex40724,

the reason why **NO** toasts were shown at all was a necessary (because of https://mantis.ilias.de/view.php?id=36967 and an issue described by Fabian Kruse in https://mantis.ilias.de/view.php?id=36967#c91762) change implemented with #5725 (by @chfsx). 

Afterwards, @iszmais made the toasts work again. The PR #5738 was merged **yesterday (late afternoon)**. With the latest state of `release_8` toasts should work again.

We noticed that the 'Who is online?' toasts could be provided without using `Services/Notifications` (they indeed misused `Services/Notifications` in the past).

Best regards
@mjansenDatabay

File: Services/Awareness/classes/Provider/AwarenessToastProvider.php
Comment (chfsx) [inline Services/Awareness/classes/Provider/AwarenessToastProvider.php]: @iszmais this ToastAction should have a non empty 'id', maybe something like `'goToUser'`

**Comment by iszmais** [comment]
Thank you for your quick reponse @alex40724 !
I really appreciate!

Aspointed out the current state of the test installation is due to another, unrelated problem wich @chfsx and myself fixed already for the release. While fixing that i stumbled upon that target issue here an worked out this proposal.

I totally agree than within the initial state of The FR of the Toast feature it would make completly sense to put the WIO Notifications inside the center. However within the implementation of the FR and the depending Jour Fixe meetings we set a goal to distinguish the Toast and Notifcation completly in Future wich comes no into reality now partialy over time.

We have now reached a point where we can create and handle Toast without the knowledge of notifications and the WIO Notifications are the mostly appropiate for this change.
Im pretty sure this change will be obligated at least for the first release patch, since it improves not only the code but has a convenient effect on the user interface too. I just wanted to anticapate the upcoming mantis ticket for it by putting it into the release, but if you feel unsure about it i totally respect your desicion to take that to the JF.

Greetings,
@iszmais

**Comment by alex40724** [comment]
@iszmais  Thanks for the explanations. I was not aware of https://github.com/ILIAS-eLearning/ILIAS/pull/5725. Maybe this larger refactoring could have been also mentioned in a Jour Fixe, since we are "almost stable". Of course this is all fine if necessary to fix the issues. Since @mjansenDatabay  and @chfsx will review this and they are both TB members, it is ok to merge this when all reviews are "green" even without JF. So I set you to be responsible to merge.

I have one open issue: The old implementation seemed to limit the number of displyed users to "20" to not retrieve to much data if a higher number of users are online. I am not sure about the performance impact but the new implementation seems not to include this limitation. Please re-add it, if you consider this being appropriate, before merging.

Thanks for taking care of adopting the new concepts in the Who-Is-Online component.

**Review by alex40724 — APPROVED** [review]

File: Services/Awareness/classes/class.WidgetManager.php
Comment (alex40724) [inline Services/Awareness/classes/class.WidgetManager.php:72]: This limitation seems to be missed in the new implementation, but may be helpful to prevent performance issues.

Comment (mjansenDatabay) [reply Services/Awareness/classes/class.WidgetManager.php:72]: Good catch Alex.

File: Services/Awareness/classes/Provider/AwarenessToastProvider.php
Comment (iszmais) [reply Services/Awareness/classes/Provider/AwarenessToastProvider.php]: I connected it to the scope awareness since this provide should handle all awareness toast atm.
If there are additional toast in future it may be usefull to split that but that for responsibles in future to decide.

File: Services/Awareness/classes/class.WidgetManager.php
Comment (iszmais) [reply Services/Awareness/classes/class.WidgetManager.php:72]: Due to the refactored Fetching this shouldnt cause any Performance issue no more, but i added anyways it for the enhanced user experience (nobody wants to see 200 new online users).

**Review by alex40724 — APPROVED** [review]

**Review by mjansenDatabay — APPROVED** [review]

**Pull request merged**
